widget/hyperlink.go

```diff
@@ -0,0 +1,140 @@
+package widget
+
+import (
+	"image/color"
+	"net/url"
+	"os/exec"
+	"runtime"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/theme"
+)
+
+// Hyperlink widget is a text component with appropriate padding and layout.
+// When clicked, the default web browser should open with a URL
+type Hyperlink struct {
+	textProvider
+	Text      string
+	Url       string
+	Alignment fyne.TextAlign // The alignment of the Text
+	TextStyle fyne.TextStyle // The style of the hyperlink text
+}
+
+// NewHyperlink creates a new layout widget with the set text content
+func NewHyperlink(text string, sUrl string) *Hyperlink {
+	// ignore the error here, we'll continue anyway
+	return NewHyperlinkWithStyle(text, sUrl, fyne.TextAlignLeading, fyne.TextStyle{})
+}
+
+// NewHyperlinkWithStyle creates a new layout widget with the set text content
+func NewHyperlinkWithStyle(text string, sUrl string, alignment fyne.TextAlign, style fyne.TextStyle) *Hyperlink {
+	hl := &Hyperlink{
+		Text:      text,
+		Url:       sUrl,
+		Alignment: alignment,
+		TextStyle: style,
+	}
+
+	return hl
+}
+
+// SetText sets the text of the hyperlink
+func (hl *Hyperlink) SetText(text string) {
+	hl.Text = text
+	hl.textProvider.SetText(text)
+	Renderer(hl).Refresh()
+}
+
+// SetUrl sets the URL of the hyperlink, taking in a string type
+func (hl *Hyperlink) SetUrl(sUrl string) {
+	hl.Url = sUrl
+}
+
+// SetUrl sets the URL of the hyperlink, taking in a Url type
```

Comment:
sorry to be picky but the type is URL, not Url like the doc says

Reply:
That's a good catch!

Reply:
Changed in 3fdc033528934b7e8ed483474c570a7f82cbcc9a